compiler/rustc_trait_selection/src/solve/mod.rs

```diff
@@ -383,6 +403,106 @@ impl<'a, 'tcx> EvalCtxt<'a, 'tcx> {
             None => self.make_canonical_response(Certainty::AMBIGUOUS),
         }
     }
+
+    fn compute_alias_eq_goal(
+        &mut self,
+        goal: Goal<'tcx, (ty::Term<'tcx>, ty::Term<'tcx>)>,
+    ) -> QueryResult<'tcx> {
+        let tcx = self.tcx();
+
+        let evaluate_normalizes_to = |ecx: &mut EvalCtxt<'_, 'tcx>, alias, rigid| {
+            ecx.infcx.probe(|_| {
+                let (_, certainty) = ecx.evaluate_goal(goal.with(
+                    tcx,
+                    ty::Binder::dummy(ty::ProjectionPredicate {
+                        projection_ty: alias,
+                        term: rigid,
+                    }),
+                ))?;
+                ecx.make_canonical_response(certainty)
+            })
+        };
+
+        if goal.predicate.0.is_infer() || goal.predicate.1.is_infer() {
+            return self.infcx.probe(|_| {
+                let nested_goals =
+                    self.infcx.eq(goal.param_env, goal.predicate.0, goal.predicate.1)?;
+                self.evaluate_all_and_make_canonical_response(nested_goals)
```

Comment:
this should be unreachable? We only emit `AliasEq` goals after we've checked for inference vars :thinking: